[BUG]Panic in buildrun controller due to nil taskRun.Status.Conditions

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Kubernetes Version

1.24

### Shipwright Version

0.15.0

### Current Behavior

The build-controller crashed with a panic caused by a nil pointer dereference.

Stack trace shows the error originates from [pkg/reconciler/buildrun/resources/conditions.go:74](https://github.com/shipwright-io/build/blob/release-v0.15/pkg/reconciler/buildrun/resources/conditions.go#L74):

```go
	case pipelineapi.TaskRunReasonTimedOut:
		reason = "BuildRunTimeout"
		message = fmt.Sprintf("BuildRun %s failed to finish within %s",
			buildRun.Name,
			taskRun.Spec.Timeout.Duration,
		)
```

```log
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x192a4cd]

github.com/shipwright-io/build/pkg/reconciler/buildrun/resources.UpdateBuildRunUsingTaskRunCondition
        at pkg/reconciler/buildrun/resources/conditions.go:74
...
controller-runtime/pkg/internal/controller.(*Controller).Reconcile
        at controller.go:111

```

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Anything else?

_No response_

--- a/pkg/reconciler/buildrun/resources/conditions.go
+++ b/pkg/reconciler/buildrun/resources/conditions.go
@@ -7,6 +7,7 @@ package resources
 import (
 	"context"
 	"fmt"
+	"time"
 
 	pipelineapi "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
 	"knative.dev/pkg/apis"
@@ -69,9 +70,16 @@ func UpdateBuildRunUsingTaskRunCondition(ctx context.Context, client client.Clie
 
 	case pipelineapi.TaskRunReasonTimedOut:
 		reason = "BuildRunTimeout"
+		var timeout time.Duration
+		if taskRun.Spec.Timeout == nil {
+			// if the TaskRun does not have a timeout set, we cannot use it to determine the BuildRun timeout
+			timeout = time.Since(taskRun.CreationTimestamp.Time)
+		} else {
+			timeout = taskRun.Spec.Timeout.Duration
+		}
 		message = fmt.Sprintf("BuildRun %s failed to finish within %s",
 			buildRun.Name,
-			taskRun.Spec.Timeout.Duration,
+			timeout,
 		)
 
 	case pipelineapi.TaskRunReasonSuccessful:

